calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -19,63 +19,100 @@ import (
 
 	"fmt"
 	"reflect"
+	"strings"
 
 	"io/ioutil"
 	"os"
 
-	"github.com/tigera/libcalico-go/lib/api"
 	"github.com/tigera/libcalico-go/lib/api/unversioned"
 
+	"bytes"
+
 	"github.com/ghodss/yaml"
 	"github.com/golang/glog"
+	"github.com/tigera/libcalico-go/lib/client"
 	"github.com/tigera/libcalico-go/lib/validator"
 )
 
+// The ResourceManager interface provides useful function for each resource type.  This includes:
+//	-  Commands to assist with generation of table output format of resources
+//	-  Commands to manage resource instances through an un-typed interface.
+type ResourceManager interface {
+	GetTableDefaultHeadings(wide bool) []string
+	GetTableTemplate(columns []string) (string, error)
+	Apply(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error)
+	Create(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error)
+	Update(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error)
+	Delete(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error)
+	List(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error)
+}
+
+type ResourceActionCommand func(*client.Client, unversioned.Resource) (unversioned.Resource, error)
+
 // ResourceHelper encapsulates details about a specific version of a specific resource:
 //
 // 	-  The type of resource (Kind and Version).  This includes the list types (even
 //	   though they are not strictly resources themselves).
 // 	-  The concrete resource struct for this version
+//	-  Template strings used to format output for each resource type.
+//	-  Functions to handle resource management actions (apply, create, update, delete, list).
+//         These functions are an untyped interface (generic Resource interfaces) that map through
+//         to the Calicc clients typed interface.
 type resourceHelper struct {
```

Comment:
Not sure if this is the right place.. but we need a way of specifying sorting behavior when listing resources.

For example, tiers are currently listed alphabetically, but it would be really nice if they were sorted by their order instead.